### ISSUE TITLE
Pass upsertGraphAndFetch an array of ids for a relation to be related

### ISSUE THREAD

**Issue by asadakbar** [Vincit/objection.js#811]
My apologies if this is somewhere in the documentation. I have scoured through them and looked at SO questions hoping to find the answer before coming here. 

TLDR; Basically I'd like to be able to pass upsertGraphAndFetch an object with an array for a relation and have the join records be created with the elements of the array treated as ids. Is there a way to do that?

I have a react app that has a key in state which maps to a relationship in my api. For example 
```
{ product: 
     { name: "something", coupons: [12, 15] },
   coupons:
     [{ id: 12, name: "Save a lot" }, { id: 15, name: "Time to save" }]
}
```

Coupons has an array of coupon ids. This was done to normalize the data. I'm currently sending state to the api and using upsertGraphAndFetch to save and relate the data. This is my product class:

```
class Product extends Model {
  static get relationMappings() {
    return {
      coupons: {
        relation: Model.ManyToManyRelation,
        modelClass: Coupon,
        join: {
          from: 'products.id',
          through: {
            from: 'product_coupons.product_id',
            to: 'product_coupons.coupon_id',
           },
           to: 'coupons.id'
         }
       }
     }
   }
 }
```
Everything was saving fine until I tried to pass the coupons array. It is erroring. From the documentation here it seemed to indicate that upsertGraph could take an array as the graph type. Maybe that doesnt work for nested keys in the json?

I'm trying to avoid having to loop through all the ids and create full objects for each of them.

Basically I'd like to be able to pass an array for a relation and have the join records be created with the elements of the array treated as ids. Is there a way to do that? Thanks so much for this amazing tool.

**Comment by koskimas** [comment]
You need to pass an array of objects instead of array of identifiers:

```js
[{id: 12}, {id: 15}]
```

insted of 

```js
[12, 15]
```

**Comment by asadakbar** [comment]
Understood. Thank you.
